test/specs/core.controller.tests.js

```diff
@@ -693,6 +693,41 @@ describe('Chart', function() {
 			chart.update();
 			expect(chart.tooltip._options).toEqual(jasmine.objectContaining(newTooltipConfig));
 		});
+
+		it ('should update the metadata', function() {
+			var cfg = {
+				data: {
+					labels: ['A', 'B', 'C', 'D'],
+					datasets: [{
+						type: 'line',
+						data: [10, 20, 30, 0]
+					}]
+				},
+				options: {
+					responsive: true,
+					scales: {
+						xAxes: [{
+							type: 'time'
+						}],
+						yAxes: [{
+							scaleLabel: {
+								display: true,
+								labelString: 'Value'
+							}
+						}]
+					}
+				}
+			};
+			var chart = acquireChart(cfg);
+			var meta = chart.getDatasetMeta(0);
+			expect(meta.type).toBe('line');
+
+			// change the dataset to bar and check that meta was updated
+			cfg.data.datasets[0].type = 'bar'
+			chart.update(cfg);
```

Comment:
`update` doesn't accept the chart config as parameter but an update config object (`{duration, easing, lazy}`)

I think I saw the same error in the financial example.